src/components/LibraryItem.tsx

```diff
@@ -174,19 +177,42 @@ export class LibraryItem extends React.Component<LibraryItemProps, LibraryItemSt
             }
         }
 
+        // Show itemSummary only if this is a leaf item and showItemSummary is set to true
+        if (this.props.showItemSummary && this.props.data.childItems.length == 0) {
+            expandIcon = (
+                <div className="ItemSummaryExpandIcon">
+                    <i className="fa fa-ellipsis-h" aria-hidden="true" onClick={this.onExpandIconClicked.bind(this)} />
+                </div>
+            );
+
+            if (this.state.itemSummaryExpanded) {
+                itemSummary = <ItemSummary
+                    libraryContainer={this.props.libraryContainer}
+                    data={this.props.data}
+                    showDescription={true}
+                />;
+            }
+        }
+
```

Comment:
Now that there's a conflict, if you can refactor this part into `getItemSummary` method that'll be awesome.